sdk/core/Azure.Core.Experimental/src/CloudMachine.cs

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Text.Json;
+using Microsoft.Extensions.Configuration;
+
+namespace Azure
+{
+    /// <summary>
+    /// Stores configuration of a CloudMachine.
+    /// </summary>
+    public class CloudMachine
+    {
+        /// <summary>
+        /// Unique identifier of a CloudMachine. It's the name of the resource group of all the CloudMachine resources.
+        /// </summary>
+        public string Id { get; private set; }
+
+        /// <summary>
+        /// Friendly name of CloudMachine. It's stored as a Tag of all Azure resources associated with the machine.
+        /// </summary>
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// Azure subscription ID.
+        /// </summary>
+        public string SubscriptionId { get; private set; }
+
+        /// <summary>
+        /// Azure region, e.g. westus2
+        /// </summary>
+        public string Region { get; private set; }
+
+        private int Version { get; set; }
+
+        /// <summary>
+        /// Creates a new CloudMachine
+        /// </summary>
+        /// <param name="subscriptionId"></param>
+        /// <param name="region"></param>
+        /// <returns></returns>
+        /// <remarks>DisplayName is initialized to id. It can be changed by setting the DisplayName property.</remarks>
+        public static CloudMachine Create(string subscriptionId, string region)
+        {
+            if (string.IsNullOrEmpty(subscriptionId)) throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrEmpty(region)) throw new ArgumentNullException(nameof(region));
+
+            var id = GenerateCloudMachineId();
+            var defaultDisplayName = $"{id}@{DateTime.UtcNow:d}";
+            var cm = new CloudMachine(id, defaultDisplayName, subscriptionId, region, 1);
+            return cm;
+        }
+
+        /// <summary>
+        /// Loads CloudMachine settings from configurationFile
+        /// </summary>
+        /// <param name="configurationFile"></param>
+        /// <exception cref="InvalidCloudMachineConfigurationException"></exception>
+        public CloudMachine(string configurationFile = ".\\cloudconfig.json")
```

Comment:
The path will be different cross platform - should we default to null so that we can do the appropriate platform agnostic path string?

Reply:
you mean like Linux and Windows?

Reply:
yeah - it would be a different path separator '/'

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.io.path.pathseparator?view=net-7.0

Reply:
fixed